src/Crowdsale/DynamicPriceSale.sol

```diff
@@ -27,9 +27,9 @@ contract DynamicSale is ContextUpgradeable, ReentrancyGuardUpgradeable, Ownable2
     uint256 public saleHardCap;
     address public treasury;
     /// @dev mininum value for usable part of the curve
-    uint256 public priceCurveMinValue = 4109 ether;
```

Comment:
You can not add constants in an upgradable contract: https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#avoid-initial-values-in-field-declarations

Reply:
What they recommend is not to set constants in the constructor or as a contract constant they won't be upgrdded, i.e., like this:

```
contract MyContract {
    uint256 public hasInitialValue = 42; // equivalent to setting in the constructor
}
```

Instead, they recommend doing it in the initializer which is the current case in this contract